cogs/gaminator.py

```diff
@@ -0,0 +1,151 @@
+logger = logging.getLogger('voluspa.cog.gaminator')
+
+def role_dict_list_to_role_ping_list(role_dict_list):
+    role_names = []
+
+    for d in role_dict_list:
+        role_names.append('`@' + d['role-name'] + '`')
+
+    return role_names
+
+def get_current_page_dict(page):
+    ret = {}
+    for i in range(len(page)):
+        ret[ri_at_index(i)] = {'role-name': list(page)[i], 'qualified-name': page[list(page)[i]]}
+
+    return ret
+
+async def set_page(current_page_dict, menu_msg, current_page_num, num_pages):
+    menu_description = ''
+    for k, v in current_page_dict.items():
+        menu_description += k + ' - `@' + v['role-name'] + '` - ' + v['qualified-name'] + '\n'
+
+    menu_embed = default_embed(title=f'Other Games {current_page_num + 1}/{num_pages}', description=menu_description)
+
+    await menu_msg.clear_reactions()
+    await menu_msg.edit(embed=menu_embed)
+
+    if current_page_num > 0:
+        await menu_msg.add_reaction('\u2b05') # Left arrow
+
+    for k in current_page_dict.keys():
+        await menu_msg.add_reaction(k)
+
+    if current_page_num < num_pages - 1:
+        await menu_msg.add_reaction('\u27a1') # Right arrow
+
+    await menu_msg.add_reaction('\u2705') # Check mark
+
+def page_dict_as_lines(d, max_lines=EMBED_MAX_LINES):
+    if max_lines > 17:
+        max_lines = 17
+    elif max_lines < 2:
+        max_lines = 2
+
+    pages = []
+    current_page = {}
+    current_line = 0
+    for k, v in d.items():
+        if current_line % max_lines == 0 and current_line > 0:
+            pages.append(current_page)
+            current_page = {}
+        current_page[k] = v
+        current_line += 1
+
+    if len(current_page) > 0:
+        pages.append(current_page)
+
+    return pages
+
+def format_roles_dict(raw):
+    ret = {}
+
+    for k, v in raw.items():
+        ret[k] = v[0].title()
+
+    return ret
+
+class Gaminator(commands.Cog):
+    def __init__(self, bot):
+        self.bot = bot
+
+    @commands.command(name='other-games')
+    async def other_games(self, ctx):
+        logger.info(f'{ctx.message.author} called other_games')
+        formatted_roles_dict = format_roles_dict(ROLES['other_games'])
+        pages = page_dict_as_lines(formatted_roles_dict)
+
+        current_page_num = 0
+        num_pages = len(pages)
+        current_page_dict = get_current_page_dict(pages[current_page_num]) # Gets a dict indexed by emoji
+
+        menu_embed = default_embed(title=f'Other Games {current_page_num + 1}/{num_pages}')
+        role_embed = default_embed(title='Your Roles')
+        role_embed.add_field(name='Adding', value='None')
+        role_embed.add_field(name='Removing', value='None')
+
+        role_msg = await ctx.send(embed=role_embed)
+        menu_msg = await ctx.send(embed=menu_embed)
+        await set_page(current_page_dict, menu_msg, current_page_num, num_pages)
+
+        roles_to_add = []
+        roles_to_remove = []
+
+        try:
+            while True:
```

Comment:
Just to verify, the Discord command timeout handles this loop structure?

Reply:
Hmm no, the timeout is handled within the loop at line 106. Is there a decorator to set a command scope timeout? I couldn't find one.